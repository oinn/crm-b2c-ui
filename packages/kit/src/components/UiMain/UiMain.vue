<template>
  <main class="ui-main" :style="styles">
    <slot></slot>
  </main>
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import { Offsets } from '@/index';

let uiListener: symbol | undefined;

export default defineComponent({
  name: 'UiMain',
  data() {
    return {
      offsets: this.$ui.uiMainOffsets,
    };
  },
  computed: {
    styles() {
      return {
        margin: `${this.offsets.top}px ${this.offsets.right}px ${this.offsets.bottom}px ${this.offsets.left}px`,
      };
    },
  },
  created() {
    uiListener = this.$ui.addListener('uiMainOffsets', (uiMainOffsets: Offsets) => {
      this.offsets = uiMainOffsets;
    });
  },
  beforeUnmount() {
    if (uiListener) {
      this.$ui.removeListener(uiListener);
    }
  },
});
</script>

<style lang="scss">
// noinspection CssUnknownTarget
@import "@/styles/config.scss";
// noinspection CssUnknownTarget
@import "@/styles/colors/index.scss";

.ui-main {
  padding: 5 * $module;
  display: flex;
  flex-direction: column;
  flex-grow: 1;
}
</style>
