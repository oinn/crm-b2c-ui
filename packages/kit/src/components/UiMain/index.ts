export { default as UiMain } from './UiMain.vue';
