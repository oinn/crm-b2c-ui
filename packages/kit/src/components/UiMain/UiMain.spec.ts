import { config, mount } from '@vue/test-utils';
import { uiProperties } from '@/index';
import UiMain from './UiMain.vue';

describe('UiToolbarTitle', () => {
  it('Rendered correctly', () => {
    const wrapper = mount(UiMain, {
      slots: {
        default: 'Default slot text',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Reading uiMainOffsets global property', () => {
    config.global.mocks = {
      $ui: {
        ...uiProperties(),
        uiMainOffsets: {
          top: 48,
          right: 56,
          bottom: 36,
          left: 256,
        },
      },
    };
    const wrapper = mount(UiMain, {
      slots: {
        default: 'Default slot text',
      },
    });

    expect(wrapper.attributes().style).toBe('margin: 48px 56px 36px 256px;');
  });
});
