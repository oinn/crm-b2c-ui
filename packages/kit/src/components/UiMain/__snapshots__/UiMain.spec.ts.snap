// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiToolbarTitle Rendered correctly 1`] = `"<main class=\\"ui-main\\" style=\\"margin: 0px 0px 0px 0px;\\">Default slot text</main>"`;
